api4/channel.go

```diff
@@ -751,6 +751,12 @@ func searchChannelsForTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 	if c.App.SessionHasPermissionToTeam(c.App.Session, c.Params.TeamId, model.PERMISSION_LIST_TEAM_CHANNELS) {
 		channels, err = c.App.SearchChannels(c.Params.TeamId, props.Term)
 	} else {
+		// If the user is not a team member, return a 404
+		if _, err = c.App.GetTeamMember(c.Params.TeamId, c.App.Session.UserId); err != nil {
```

Comment:
I've added a check to avoid going to the database if the user is not a member of the team, assuming that, if you don't have permissions to list channels nor belong to the team, it doesn't make sense to perform the search. Please let me know if there are other cases that I'm not considering, or there is a preferred way to manage this flow